openghg/plotting/_timeseries.py

```diff
@@ -1,61 +1,191 @@
-from typing import List, Optional, Union
-
+from typing import List, Optional, Union, Dict, Tuple
 import plotly.graph_objects as go
+import numpy as np
+import base64
+
 from openghg.dataobjects import ObsData
+from openghg.util import get_species_info, synonyms, get_datapath
+
+
+def _latex2html(
+    latex_string: str
+) -> str:
+    """Replace latex sub/superscript formatting with html.
+    Written because the latex formatting in Plotly seems inconsistent
+    (works in Notebooks, but not VC Code at the moment).
+
+    Args:
+        latex_string: 
+    """
+
+    replacements = {"$^2$": "<sup>2</sup>",
+                    "$^{-1}$": "<sup>-1</sup>",
+                    "$^{-2}$": "<sup>-2</sup>",
+                    "$_2$": "<sub>2</sub>",
+                    "$_3$": "<sub>3</sub>",
+                    "$_4$": "<sub>4</sub>",
+                    "$_5$": "<sub>5</sub>",
+                    "$_6$": "<sub>6</sub>",
+                    }
+
+    html_string = latex_string
+    for rep in replacements:
+        html_string = html_string.replace(rep,
+                                        replacements[rep])
+
+    return html_string
+
+
+def _plot_remove_gaps(
+        x_data: np.ndarray,
+        y_data: np.ndarray,
+        gap: Optional[int] = 24 * 60 * 60 * 1000000000
+) -> Tuple[np.ndarray, np.ndarray]:
+    '''Insert NaNs between big gaps in the data. 
+    Prevents connecting lines being drawn
+
+    Args:
+        x_data: plot timeseries (numpy timestamp)
+        y_data: data array
+        gap: gap beyond which a NaN is introducted (nanoseconds, default=1 day)
+
+    Returns:
+        x, y: x and y arrays to plot
+    '''
+
+    gap_idx = np.where(np.diff(x_data.astype(int)) > gap)[0]
+    x_data_plot = np.insert(x_data, gap_idx + 1, values=x_data[0])
+    y_data_plot = np.insert(y_data, gap_idx + 1, values=np.nan)
+
+    return x_data_plot, y_data_plot
+
+
+def _plot_legend_position(
+        ascending: bool
+) -> Tuple[Dict, Dict]:
+    """Position of legend and logo,
+    depending on whether data is ascending or descending
+
+    Args:
+        ascending (bool): Is the data ascending
```

Comment:
Here we don't need the type after the variable name as we're using type hints.

Reply:
```suggestion
        ascending: Is the data ascending
```